osu.Game.Rulesets.Mania/Difficulty/ManiaDifficultyCalculator.cs

```diff
@@ -30,14 +30,15 @@ public ManiaDifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beatmap, Mod[] mods, Skill[] skills, double clockRate)
         {
             if (beatmap.HitObjects.Count == 0)
-                return new ManiaDifficultyAttributes { Mods = mods };
+                return new ManiaDifficultyAttributes { Mods = mods, Skills = skills };
```

Comment:
might make sense to restructure the code here so the common portion of population isn't done twice (invert the if condition to only run the population if available)?

Reply:
Everywhere or just for mania? I've tried this for OsuDifficultyCalculator and I don't know if it is an improvement or not.

```cs
protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beatmap, Mod[] mods, Skill[] skills, double clockRate)
{
    var attr = new OsuDifficultyAttributes { Mods = mods, Skills = skills };

    if (beatmap.HitObjects.Count == 0)
        return attr;

    attr.AimStrain = Math.Sqrt(skills[0].DifficultyValue()) * difficulty_multiplier;
    attr.SpeedStrain = Math.Sqrt(skills[1].DifficultyValue()) * difficulty_multiplier;
    attr.StarRating = attr.AimStrain + attr.SpeedStrain + Math.Abs(attr.AimStrain - attr.SpeedStrain) / 2;

    // Todo: These int casts are temporary to achieve 1:1 results with osu!stable, and should be removed in the future
    double hitWindowGreat = (int)(beatmap.HitObjects.First().HitWindows.Great / 2) / clockRate;
    double preempt = (int)BeatmapDifficulty.DifficultyRange(beatmap.BeatmapInfo.BaseDifficulty.ApproachRate, 1800, 1200, 450) / clockRate;
    attr.ApproachRate = preempt > 1200 ? (1800 - preempt) / 120 : (1200 - preempt) / 150 + 5;

    attr.OverallDifficulty = (80 - hitWindowGreat) / 6;

    attr.MaxCombo = beatmap.HitObjects.Count;
    // Add the ticks + tail of the slider. 1 is subtracted because the head circle would be counted twice (once for the slider itself in the line above)
    attr.MaxCombo += beatmap.HitObjects.OfType<Slider>().Sum(s => s.NestedHitObjects.Count - 1);

    return attr;
}
```